app/config/config.yml

```diff
@@ -71,3 +73,42 @@ swiftmailer:
     spool:     { type: memory }
 
 fos_http_cache: ~
+
+fos_rest:
+    routing_loader:
+        default_format: json
+    view:
+        formats:
+            json: true
+        view_response_listener: 'force'
+    format_listener:
+        rules:
+            - { path: '^/api', priorities: ['json'], fallback_format: json, prefer_extension: true }
+            - { path: '^/', stop: true }
+    exception:
+        codes:
+            'Symfony\Component\Routing\Exception\ResourceNotFoundException': 404
+        messages:
+            'Symfony\Component\Routing\Exception\ResourceNotFoundException': true
+
+sensio_framework_extra:
+    view:    { annotations: false }
+
+swp_updater:
+    version_class: %swp_updater.version.class%
+    monolog_channel: true
+    client:
+        base_uri: http://rafal.sourcefabric.net
```

Comment:
With Guzzle you can also specify in the options to resolve a url differently. Check the readme of my bundle, it allows to set default options for every request, with the following config you can achieve this.

curl:
    10203: # CURLOPT_RESOLVE
         - "example.com:5050:localhost"  # This will resolve the host example.com to your localhost